includes/tuque_wrapper.inc

```diff
@@ -240,7 +240,15 @@ class IslandoraFedoraObject extends FedoraObject {
       return $ret;
     }
     catch (Exception $e) {
-      watchdog('islandora', 'Failed to ingest datastream @datastream on object: @pid</br>code: @code<br/>message: @msg', array(
+      if ($e instanceof DatastreamExistsException) {
+        $message = islandora_deprecated('7.x-1.10', t('The datastream @dsid already exists on the object: @pid. After the 7.x-1.10 release this will throw an exception.', array(
```

Comment:
I'm not sure the use of `islandora_deprecated()` is entirely appropriate here...

Little script forcing it to be emitted (just ran in `devel/php`):
```php
$object = islandora_object_load('islandora:root');
$ds = $object->constructDatastream('DC', 'M');
$object->ingestDatastream($ds);
```
yields the message:

> User deprecated function: ingestDatastream() has been deprecated. As of 7.x-1.10, please update your code before the next release. The datastream DC already exists on the object: islandora:root. After the 7.x-1.10 release this will throw an exception. in trigger_error() (line 248 of /var/www/drupal7/sites/all/modules/islandora/includes/tuque_wrapper.inc).

... Asserting that the `ingestDatastream()` method itself is deprecated (and having weird language besides). Also, gets into the whole unable-to-have-fully-updated-code-work-appropriately business because it has no means to receive the exception, essentially requiring a secondary deprecation pass wherever things are updated to handle it... this is why I was [thinking to introduce a variable to facilitate things](https://github.com/Islandora/islandora/pull/680#discussion_r120157983)... to allow the exception to the thrown as it will be after the eventual release... Admittedly, doesn't seem ideal, but _a_ way it might be addressed...

Really, the big thing in my mind is that there's no way to suppress the message after updating your code, without hacking it out of here...